apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -0,0 +1,1234 @@
+
+    %% Check that Bert got it.
+    GBO2Arg = args_to_binary([BPubkey]),
+    GBO2Ret = call_func(NodeName, APubkey, APrivkey, EncodedC1Pubkey,
+			HexCode, <<"get_balance_of">>, GBO2Arg),
+    #{<<"value">> := 2015000} = decode_data(<<"int">>, GBO2Ret),
+
+    %% Spend 6000 explicitly from contract 1 to Bert.
+    SF1Arg = args_to_binary([DecodedC1Pubkey,BPubkey,6000]),
+    SF1Ret = call_func(NodeName, APubkey, APrivkey, EncodedC2Pubkey,
+		       HexCode, <<"spend_from">>, SF1Arg),
+    #{<<"value">> := 2021000} = decode_data(<<"int">>, SF1Ret),
+
+    aecore_suite_utils:mine_blocks(NodeName, 3),
+
+    %% Check that Bert got it.
+    GBO3Arg = args_to_binary([BPubkey]),
+    GBO3Ret = call_func(NodeName, APubkey, APrivkey, EncodedC1Pubkey,
+			HexCode, <<"get_balance_of">>, GBO3Arg),
+    #{<<"value">> := 2021000} = decode_data(<<"int">>, GBO3Ret),
+
+    %% Check contract 2 balance.
+    GBO4Arg = args_to_binary([DecodedC2Pubkey]),
+    GBO4Ret = call_func(NodeName, APubkey, APrivkey, EncodedC1Pubkey,
+			HexCode, <<"get_balance_of">>, GBO4Arg),
+    #{<<"value">> := 5000} = decode_data(<<"int">>, GBO4Ret),
+
+    ok.
+
+%% dutch_auction_contract(Config)
+%%  Check the DutchAuction contract. We use 3 accounts here, Alice for
+%%  setting up the account, Carl as beneficiary and Bert as
+%%  bidder. This makes it a bit easier to keep track of the values as
+%%  we have gas loses as well.
+
+dutch_auction_contract(Config) ->
+    NodeName = proplists:get_value(node_name, Config),
+    %% Get account information.
+    #{acc_a := #{pub_key := APubkey,
+		 priv_key := APrivkey},
+      acc_b := #{pub_key := BPubkey,
+		 priv_key := BPrivkey},
+      acc_c := #{pub_key := CPubkey}} = proplists:get_value(accounts, Config),
+
+    %% Make sure accounts have enough tokens.
+    ABal0 = ensure_balance(APubkey, 500000),
+    BBal0 = ensure_balance(BPubkey, 500000),
+    CBal0 = ensure_balance(CPubkey, 500000),
+    {ok,[_]} = aecore_suite_utils:mine_blocks(NodeName, 1),
+
+    %% Compile contract "dutch_auction.aes"
+    ContractString = aeso_test_utils:read_contract("dutch_auction"),
+    BinCode = aeso_compiler:from_string(ContractString, []),
+    HexCode = aeu_hex:hexstring_encode(BinCode),
+
+    %% Initialise contract owned by Alice with Carl as benficiary.
+    InitArgument = args_to_binary([CPubkey,50000,500]),
+    {EncodedContractPubkey,_,_} = create_contract(NodeName, APubkey, APrivkey,
+						  HexCode, InitArgument),
+
+    ABal1 = get_balance(APubkey),
+    BBal1 = get_balance(BPubkey),
+    CBal1 = get_balance(CPubkey),
+
+    %% Mine 10 times to decrement value by 5000.
+    {ok,_} = aecore_suite_utils:mine_blocks(NodeName, 10),
+
+    %% Call the contract bid function by Bert.
+    BidValue = call_func(NodeName, BPubkey, BPrivkey, EncodedContractPubkey,
+			 HexCode, <<"bid">>, <<"()">>, #{amount => 100000}),
+    ct:pal("Bid value ~p\n", [BidValue]),
+
+    %% These values look reasonable.
```

Comment:
Could you please clarify whether the values of the balances are meant to be part of the passed/failed outcome of the test case? See also https://github.com/aeternity/epoch/wiki/Testing-general#debugging

Reply:
This should have been the same type of test as in spending_1 but the gas complicated comparing the values. This will be fixed.